### ISSUE TITLE
[Feature Request] Battery Percentage

### ISSUE THREAD

**Issue by DrJedikiah** [SpriteOvO/AirPodsDesktop#4]
I noticed that battery percentage on similar apps are not precise, it's always like 50% 60% not like 58% incrementing by 10.

At first I thought it might be something related to AWDL hardware but my mac os installed laptop which has an intel bluetooth shows the exact precise battery percentage, so it's completely about software.

Do you have any idea how macOS or iOS gathers exact battery info?

**Comment by SpriteOvO** [comment]
Because the unencrypted part (if it is not a hash) of the BLE advertisement packet broadcast by AirPods, it seems to contain only battery values in steps of 10% (only 4 bits).

https://github.com/SpriteOvO/AirPodsDesktop/blob/09e19cfdefc0eb23351fc606e2b39212ccc5c0d2/Source/Core/AppleCP.h#L144-L145

As for the units digit, some people think it is faked by iOS or macOS, and I think it is also possible that it is contained in the encrypted part, or communicated in some other way.

In the future, I'll try to start a timer when the AirPods first battery changes and also do a fake units digit value with time, I want it to change in 1% steps, but if it doesn't work well, I might make it change in 5% steps.

If anyone has a better idea, feel free to comment here or open a PR.

**Comment by DrJedikiah** [comment]
I can ensure you it's not faked. Maybe reversing macOS' IOBluetoothFamily would help but I'm not that familiar with macOS.

**Comment by SpriteOvO** [comment]
I'm also not good at macOS, I've never even used it, so I have no idea about the reverse of that system :/

**Comment by UltimateTanA** [comment]
Yes even i think if you can make the battery info more precise by making the increments of 5 or even 1 if possible
that'd be really nice
